### ISSUE TITLE
v2.32.3 - Unable to start container when Syslog Standard destination set in ENV

### ISSUE THREAD

**Issue by davidbattyJDS** [splunk/splunk-connect-for-syslog#1754]
The following error is seen on startup:

```
Jul 18 08:57:10 sc4s systemd[1]: Starting SC4S Container...
Jul 18 08:57:10 sc4s systemd[1]: Started SC4S Container.
Jul 18 08:57:11 sc4s docker[3772]: SC4S_ENV_CHECK_HEC: Splunk HEC connection test successful to index=main for sourcetype=sc4s:fallback...
Jul 18 08:57:11 sc4s docker[3772]: SC4S_ENV_CHECK_HEC: Splunk HEC connection test successful to index=prd_monitoring for sourcetype=sc4s:events...
Jul 18 08:57:12 sc4s docker[3772]: Traceback (most recent call last):
Jul 18 08:57:12 sc4s docker[3772]:   File "/etc/syslog-ng/conf.d/destinations/dest_syslog/plugin.py", line 76, in <module>
Jul 18 08:57:12 sc4s docker[3772]:     transport=transport,
Jul 18 08:57:12 sc4s docker[3772]: NameError: name 'transport' is not defined
Jul 18 08:57:12 sc4s docker[3772]: [2022-07-17T22:57:12.278952] confgen: Generator program returned with non-zero exit code; reference='/etc/syslog-ng/conf.d/destinations/dest_syslog/plugin.conf:3:1', context='root', block='dest_syslog', exec='/etc/syslog-ng/conf.d/destinations/dest_syslog/plugin.py', rc='256'
Jul 18 08:57:12 sc4s docker[3772]: Error parsing config, syntax error, unexpected LL_ERROR, expecting end of file in /etc/syslog-ng/conf.d/destinations/dest_syslog/plugin.conf:3:1-3:12:
Jul 18 08:57:12 sc4s docker[3772]: Included from /etc/syslog-ng/syslog-ng.conf:49:1-49:1:
Jul 18 08:57:12 sc4s docker[3772]: 44      @include "conf.d/local/config/sources/*.conf"
Jul 18 08:57:12 sc4s docker[3772]: 45      @include "conf.d/local/config/sources/*/*.conf"
Jul 18 08:57:12 sc4s docker[3772]: 46
Jul 18 08:57:12 sc4s docker[3772]: 47      @include "conf.d/destinations/*.conf"
Jul 18 08:57:12 sc4s docker[3772]: 48      @include "conf.d/destinations/*/*.conf"
Jul 18 08:57:12 sc4s docker[3772]: 49---->
Jul 18 08:57:12 sc4s docker[3772]: 49----> ^
Jul 18 08:57:12 sc4s docker[3772]: 50      @include "conf.d/local/config/destinations/*.conf"
Jul 18 08:57:12 sc4s docker[3772]: 51      @include "conf.d/local/config/destinations/*/*.conf"
Jul 18 08:57:12 sc4s docker[3772]: 52
Jul 18 08:57:12 sc4s docker[3772]: 53      # Group 0 log paths run before all others for high level activity like null queue
Jul 18 08:57:12 sc4s docker[3772]: 54      @include "conf.d/log_paths/0/*/*.conf"
Jul 18 08:57:12 sc4s docker[3772]: syslog-ng documentation: https://www.syslog-ng.com/technical-documents/list/syslog-ng-open-source-edition
Jul 18 08:57:12 sc4s docker[3772]: contact: https://lists.balabit.hu/mailman/listinfo/syslog-ng
Jul 18 08:57:13 sc4s docker[3772]: Traceback (most recent call last):
Jul 18 08:57:13 sc4s docker[3772]:   File "/etc/syslog-ng/conf.d/destinations/dest_syslog/plugin.py", line 76, in <module>
Jul 18 08:57:13 sc4s docker[3772]:     transport=transport,
Jul 18 08:57:13 sc4s docker[3772]: NameError: name 'transport' is not defined
Jul 18 08:57:13 sc4s docker[3772]: [2022-07-17T22:57:13.199104] confgen: Generator program returned with non-zero exit code; reference='/etc/syslog-ng/conf.d/destinations/dest_syslog/plugin.conf:3:1', context='root', block='dest_syslog', exec='/etc/syslog-ng/conf.d/destinations/dest_syslog/plugin.py', rc='256'
Jul 18 08:57:13 sc4s docker[3772]: Error parsing config, syntax error, unexpected LL_ERROR, expecting end of file in /etc/syslog-ng/conf.d/destinations/dest_syslog/plugin.conf:3:1-3:12:
Jul 18 08:57:13 sc4s docker[3772]: Included from /etc/syslog-ng/syslog-ng.conf:49:1-49:1:
Jul 18 08:57:13 sc4s docker[3772]: 44      @include "conf.d/local/config/sources/*.conf"
Jul 18 08:57:13 sc4s docker[3772]: 45      @include "conf.d/local/config/sources/*/*.conf"
Jul 18 08:57:13 sc4s docker[3772]: 46
Jul 18 08:57:13 sc4s docker[3772]: 47      @include "conf.d/destinations/*.conf"
Jul 18 08:57:13 sc4s docker[3772]: 48      @include "conf.d/destinations/*/*.conf"
Jul 18 08:57:13 sc4s docker[3772]: 49---->
Jul 18 08:57:13 sc4s docker[3772]: 49----> ^
Jul 18 08:57:13 sc4s docker[3772]: 50      @include "conf.d/local/config/destinations/*.conf"
Jul 18 08:57:13 sc4s docker[3772]: 51      @include "conf.d/local/config/destinations/*/*.conf"
Jul 18 08:57:13 sc4s docker[3772]: 52
Jul 18 08:57:13 sc4s docker[3772]: 53      # Group 0 log paths run before all others for high level activity like null queue
Jul 18 08:57:13 sc4s docker[3772]: 54      @include "conf.d/log_paths/0/*/*.conf"
Jul 18 08:57:13 sc4s docker[3772]: syslog-ng documentation: https://www.syslog-ng.com/technical-documents/list/syslog-ng-open-source-edition
Jul 18 08:57:13 sc4s docker[3772]: contact: https://lists.balabit.hu/mailman/listinfo/syslog-ng
Jul 18 08:57:13 sc4s docker[3772]: syslog-ng checking config
Jul 18 08:57:13 sc4s docker[3772]: sc4s version=2.32.3
Jul 18 08:57:14 sc4s docker[3772]: starting goss
Jul 18 08:57:15 sc4s docker[3772]: Traceback (most recent call last):
Jul 18 08:57:15 sc4s docker[3772]:   File "/etc/syslog-ng/conf.d/destinations/dest_syslog/plugin.py", line 76, in <module>
Jul 18 08:57:15 sc4s docker[3772]:     transport=transport,
Jul 18 08:57:15 sc4s docker[3772]: NameError: name 'transport' is not defined
Jul 18 08:57:15 sc4s docker[3772]: [2022-07-17T22:57:15.108213] confgen: Generator program returned with non-zero exit code; reference='/etc/syslog-ng/conf.d/destinations/dest_syslog/plugin.conf:3:1', context='root', block='dest_syslog', exec='/etc/syslog-ng/conf.d/destinations/dest_syslog/plugin.py', rc='256'
Jul 18 08:57:15 sc4s docker[3772]: Error parsing config, syntax error, unexpected LL_ERROR, expecting end of file in /etc/syslog-ng/conf.d/destinations/dest_syslog/plugin.conf:3:1-3:12:
Jul 18 08:57:15 sc4s docker[3772]: Included from /etc/syslog-ng/syslog-ng.conf:49:1-49:1:
Jul 18 08:57:15 sc4s docker[3772]: 44      @include "conf.d/local/config/sources/*.conf"
Jul 18 08:57:15 sc4s docker[3772]: 45      @include "conf.d/local/config/sources/*/*.conf"
Jul 18 08:57:15 sc4s docker[3772]: 46
Jul 18 08:57:15 sc4s docker[3772]: 47      @include "conf.d/destinations/*.conf"
Jul 18 08:57:15 sc4s docker[3772]: 48      @include "conf.d/destinations/*/*.conf"
Jul 18 08:57:15 sc4s docker[3772]: 49---->
Jul 18 08:57:15 sc4s docker[3772]: 49----> ^
Jul 18 08:57:15 sc4s docker[3772]: 50      @include "conf.d/local/config/destinations/*.conf"
Jul 18 08:57:15 sc4s docker[3772]: 51      @include "conf.d/local/config/destinations/*/*.conf"
Jul 18 08:57:15 sc4s docker[3772]: 52
Jul 18 08:57:15 sc4s docker[3772]: 53      # Group 0 log paths run before all others for high level activity like null queue
Jul 18 08:57:15 sc4s docker[3772]: 54      @include "conf.d/log_paths/0/*/*.conf"
Jul 18 08:57:15 sc4s docker[3772]: syslog-ng documentation: https://www.syslog-ng.com/technical-documents/list/syslog-ng-open-source-edition
Jul 18 08:57:15 sc4s docker[3772]: contact: https://lists.balabit.hu/mailman/listinfo/syslog-ng
Jul 18 08:57:15 sc4s systemd[1]: sc4s.service: Main process exited, code=exited, status=1/FAILURE
Jul 18 08:57:15 sc4s systemd[1]: sc4s.service: Failed with result 'exit-code'.
```

Sample env_file config
```
# FIREEYE
SC4S_DEST_SYSLOG_PALANTIRFE_HOST=127.0.0.1
SC4S_DEST_SYSLOG_PALANTIRFE_PORT=514
SC4S_DEST_SYSLOG_PALANTIRFE_TRANSPORT=tls
SC4S_DEST_SYSLOG_PALANTIRFE_MODE=SELECT
```

**Comment by davidbattyJDS** [comment]
v2.32.4 has the following error logs

```
Jul 18 19:52:42 sc4s systemd[1]: Starting SC4S Container...
Jul 18 19:52:42 sc4s systemd[1]: Started SC4S Container.
Jul 18 19:52:42 sc4s docker[23417]: SC4S_ENV_CHECK_HEC: Splunk HEC connection test successful to index=main for sourcetype=sc4s:fallback...
Jul 18 19:52:42 sc4s docker[23417]: SC4S_ENV_CHECK_HEC: Splunk HEC connection test successful to index=prd_monitoring for sourcetype=sc4s:events...
Jul 18 19:52:45 sc4s docker[23417]: syslog-ng checking config
Jul 18 19:52:45 sc4s docker[23417]: sc4s version=2.32.4
Jul 18 19:52:46 sc4s docker[23417]: starting goss
Jul 18 19:52:47 sc4s docker[23417]: starting syslog-ng
Jul 18 19:52:49 sc4s docker[23417]: [2022-07-18T09:52:49.105526] transport(tls) was specified, but tls() options missing;
Jul 18 19:52:49 sc4s docker[23417]: [2022-07-18T09:52:49.106157] Error initializing message pipeline; plugin_name='syslog', location='root generator dest_syslog:5:5'
Jul 18 19:52:49 sc4s docker[23417]: syslog-ng failed to start; exiting...
Jul 18 19:52:49 sc4s docker[23417]: Handling exit 1 and restarting
Jul 18 19:52:49 sc4s docker[23417]: starting syslog-ng
Jul 18 19:52:51 sc4s docker[23417]: [2022-07-18T09:52:51.033791] transport(tls) was specified, but tls() options missing;
Jul 18 19:52:51 sc4s docker[23417]: [2022-07-18T09:52:51.034498] Error initializing message pipeline; plugin_name='syslog', location='root generator dest_syslog:5:5'
Jul 18 19:52:51 sc4s docker[23417]: syslog-ng failed to start; exiting...
Jul 18 19:52:51 sc4s docker[23417]: Handling exit 1 and restarting
Jul 18 19:52:51 sc4s docker[23417]: starting syslog-ng
Jul 18 19:52:53 sc4s docker[23417]: [2022-07-18T09:52:53.245578] transport(tls) was specified, but tls() options missing;
Jul 18 19:52:53 sc4s docker[23417]: [2022-07-18T09:52:53.246153] Error initializing message pipeline; plugin_name='syslog', location='root generator dest_syslog:5:5'

```

**Comment by rjha-splunk** [comment]
Its not an error for using tls options, tls flags need to be set.

**Comment by pwickham** [comment]
Please note this issue is not resolved.
We have made a PR which we believe will fix the issue. Please revirew and advise
#1755

**Comment by rjha-splunk** [comment]
Thanks @pwickham ,thank you so much contributing and yes it was overlooked in the implemented fix as i was not using tls as transport in my tests, i presumed we have edge cases sorted, let me do my testing and i will release fix today.

**Comment by rjha-splunk** [comment]
Fixed in version 2.32.5
